app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -41,38 +45,99 @@ export default class SchoolDisciplineDashboard extends React.Component {
+      if (selectedCategory && !this.filterIncidents(student.discipline_incidents, true).length) return false;
+      return true;
     });
-    return studentDisciplineIncidentCounts;
   }
 
-  sortChartKeys(groupedIncidents) {
-    const chartKeys = Object.keys(groupedIncidents);
-    switch(this.state.selectedChart) {
-    case 'time': return this.sortedTimes(chartKeys);
-    case 'day': return this.sortedDays(chartKeys);
-    case 'grade': return this.sortedGrades(chartKeys);
-    default: return this.sortedByIncidents(groupedIncidents); //because number of incidents in each category needed here
+  timeStampToHour(incident) {
+    return incident.has_exact_time ? moment.utc(incident.occurred_at).startOf('hour').format('h:mm a') : "Not Logged";
+  }
+
+  timeStampToDay(incident) {
+    return moment.utc(incident.occurred_at).format("ddd");
+  }
+
+  getIncidentsFromStudents(students) {
+    //for chart data, does not filter on selected categories
+    return _.flatten(students.map(student => this.filterIncidents(student.discipline_incidents, false)));
+  }
+
+  //Depending on the chart, incidents are grouped either by student attribute or incident attribute. Any future
+  //charts may be handled here.
+  getChartData(students, group) {
+    switch(group) {
+    case 'homeroom_label': case 'race': {
+      const groupedStudents = _.groupBy(students, group);
+      const categories = this.sortedByStudent(groupedStudents);
+      const seriesData = categories.map(category => {
+        return this.getIncidentsFromStudents(groupedStudents[category]).length;
+      });
+      return {categories, seriesData};
+    }
+    case 'grade': {
+      const groupedStudents = _.groupBy(students, group);
+      const categories = this.sortedGrades(Object.keys(groupedStudents));
+      const seriesData = categories.map(category => {
+        return this.getIncidentsFromStudents(groupedStudents[category]).length;
+      });
+      return {categories, seriesData};
+    }
+    case 'incident_code': case 'incident_location': {
+      const incidents = this.getIncidentsFromStudents(students);
+      const groupedIncidents = _.groupBy(incidents, group);
+      const categories = this.sortedByIncidents(groupedIncidents);
+      const seriesData = categories.map(category => groupedIncidents[category].length);
+      return {categories, seriesData};
+    }
+    case 'time': {
+      const incidents = this.getIncidentsFromStudents(students);
+      const groupedIncidents = _.groupBy(incidents, incident => {
+        return incident.has_exact_time ? moment.utc(incident.occurred_at).startOf('hour').format('h:mm a') : "Not Logged";
+      });
+      const categories = this.sortedTimes(Object.keys(groupedIncidents));
+      const seriesData = categories.map(category => groupedIncidents[category].length);
+      return {categories, seriesData};
+    }
+    case 'day': {
```

Comment:
If I try to use this locally, it doesn't work and I see this stack trace:

<img width="948" alt="screen shot 2018-10-02 at 10 04 49 am" src="https://user-images.githubusercontent.com/1056957/46353648-b9077580-c62a-11e8-83a8-ec53df3db221.png">